### PR TITLE
test: run all in parallel

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

Change to run all platforms in parallel.

## Why is it important?

We want to speed up the packaging process as much as possible.

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

**Comment by kuisathaverat** [comment]
nice

```
12:16:37  java.lang.RuntimeException: Method code too large!
12:16:37  	at groovyjarjarasm.asm.MethodWriter.a(Unknown Source)
12:16:37  	at groovyjarjarasm.asm.ClassWriter.toByteArray(Unknown Source)
12:16:37  	at org.codehaus.groovy.control.CompilationUnit$17.call(CompilationUnit.java:827)
12:16:37  	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1065)
12:16:37  	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:603)
12:16:37  	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:581)
12:16:37  	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:558)
12:16:37  	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:298)
12:16:37  	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:268)
12:16:37  	at groovy.lang.GroovyShell.parseClass(GroovyShell.java:688)
12:16:37  	at groovy.lang.GroovyShell.parse(GroovyShell.java:700)
12:16:37  	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.doParse(CpsGroovyShell.java:142)
12:16:37  	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.reparse(CpsGroovyShell.java:127)
12:16:37  	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.parseScript(CpsFlowExecution.java:561)
12:16:37  	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:522)
12:16:37  	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:337)
12:16:37  	at hudson.model.ResourceController.execute(ResourceController.java:97)
12:16:37  	at hudson.model.Executor.run(Executor.java:428)
12:16:37  
12:16:37  1 error
```

**Review by v1v — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Review by v1v — CHANGES_REQUESTED** [review]
I'm tempted to say this might cause certain issues with the limit of concurrent workers, since we got only 300. If we move forward, then we need to increase that number quite a lot.

For now, as long as we have issues with the google storage I'd say to hold this PR

**Review by cachedout — APPROVED** [review]

**Comment by kuisathaverat** [comment]
it spins 150 agents that take about 5 min to build its packages, but the provision time of the agents kill the improve of the change. I am going to try to spin fewer agents grouping architectures.

**Comment by EricDavisX** [comment]
excellent progress folks, thanks for the attention on it!

**Comment by kuisathaverat** [comment]
ok, we have added flags fo be able to launch different platforms and beats alone, for a simple build with only one platform the build takes less than 10 min, for all the platforms takes about 24 min

**Review by cachedout — APPROVED** [review]

**Comment by kuisathaverat** [comment]
> I'm tempted to say this might cause certain issues with the limit of concurrent workers, since we got only 300. If we move forward, then we need to increase that number quite a lot.
> 
> For now, as long as we have issues with the google storage I'd say to hold this PR

it now uses 75 Linux ephemeral workers more of then for 5-10 min

**Comment by kuisathaverat** [comment]
the concurrence issue related to GCP is resolved too

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -38,8 +39,29 @@ pipeline {
...
+    booleanParam(name: 'packetbeat', defaultValue: true, description: 'Build packetbeat binaries.')
+    booleanParam(name: 'winlogbeat', defaultValue: true, description: 'Build winlogbeat binaries.')
+    booleanParam(name: 'elastic_agent', defaultValue: true, description: 'Build elastic-agent binaries.')
+    booleanParam(name: 'dockerlogbeat', defaultValue: true, description: 'Build dockerlogbeat binaries.')
+    booleanParam(name: 'functionbeat', defaultValue: true, description: 'Build functionbeat binaries.')
+
+    booleanParam(name: 'sign_on_macos', defaultValue: false, description: 'Sign macOS binaries.')
+    booleanParam(name: 'archive_on_gcp', defaultValue: true, description: 'Enable/Disable the archive of binaries on GCP.')
+    booleanParam(name: 'push_docker', defaultValue: true, description: 'Enable/Disable the push of the Docker images.')
+    booleanParam(name: 'dry_run', defaultValue: false, description: 'Execute the pipeline without generate packages.')
```
Comment (v1v) [inline .ci/packaging.groovy:64]: Is it required to convert those params to env variables to workaround JENKINS-41929?

Comment (kuisathaverat) [reply .ci/packaging.groovy:64]: no needed, the pipeline uses the parameters not the environment variables

**Comment by v1v** [comment]
image
 74 stalled builds are affecting the Beats queue now :/

**Pull request closed without merging**